### PR TITLE
Update the contribution process

### PR DESCRIPTION
**Pull request by jin**
1) Add note about creating forks
2) Formatting
3) Change bazel-sheriff to bazel-dev
4) Don't recommend amending commits; create new commits instead.
5) s/engineer at Google/Bazel maintainer at Google"

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -21,23 +21,30 @@ here](https://join.slack.com/t/bazelbuild/shared_invite/enqtndyyodaxndy4mjqxlwu1
...
+   requests](https://help.github.com/articles/about-pull-requests/). Note that
+   we restrict permissions to create branchs on the main Bazel repository, so
+   you will need to push your commit to [your own fork of the
+   repository](https://help.github.com/articles/working-with-forks/).
+6. Wait for a Bazel team member to assign you a reviewer. It should be done in 2
+   business days (excluding holidays in the USA and Germany). If you do not get
+   a reviewer within that time frame, you can ask for one by sending a mail to
+   [bazel-dev@googlegroups.com](mailto:bazel-dev@googlegroups.com). You can also
+   assign yourself a reviewer if you know who the reviewer should be (e.g.,
+   because they reviewed an earlier related change).
```
Comment (laurentlb) [inline contributing.md]: I don't think anyone can set/change the reviewer.

File: contributing.md [hunk 2]
```diff
@@ -21,23 +21,30 @@ here](https://join.slack.com/t/bazelbuild/shared_invite/enqtndyyodaxndy4mjqxlwu1
...
+5. Send us a pull request on
+   [GitHub](https://github.com/bazelbuild/bazel/pulls). If you're new to GitHub,
+   read [about pull
+   requests](https://help.github.com/articles/about-pull-requests/). Note that
+   we restrict permissions to create branchs on the main Bazel repository, so
+   you will need to push your commit to [your own fork of the
+   repository](https://help.github.com/articles/working-with-forks/).
+6. Wait for a Bazel team member to assign you a reviewer. It should be done in 2
+   business days (excluding holidays in the USA and Germany). If you do not get
+   a reviewer within that time frame, you can ask for one by sending a mail to
```
Comment (laurentlb) [inline contributing.md]: send also an email if the review takes too long

File: contributing.md [hunk 3]
```diff
@@ -21,23 +21,30 @@ here](https://join.slack.com/t/bazelbuild/shared_invite/enqtndyyodaxndy4mjqxlwu1
...
+2. Discuss your plan and design, and get agreement on our [mailing
+   list](https://groups.google.com/forum/#!forum/bazel-dev).
+3. Prepare a git commit that implements the feature. Don't forget to add tests.
+4. Ensure you've signed a [Contributor License
+   Agreement](https://cla.developers.google.com).
+5. Send us a pull request on
+   [GitHub](https://github.com/bazelbuild/bazel/pulls). If you're new to GitHub,
+   read [about pull
+   requests](https://help.github.com/articles/about-pull-requests/). Note that
+   we restrict permissions to create branchs on the main Bazel repository, so
```
Comment (katre) [inline contributing.md]: "branchs" -> "branches"

File: contributing.md [hunk 4]
```diff
@@ -21,23 +21,30 @@ here](https://join.slack.com/t/bazelbuild/shared_invite/enqtndyyodaxndy4mjqxlwu1
...
+   repository](https://help.github.com/articles/working-with-forks/).
+6. Wait for a Bazel team member to assign you a reviewer. It should be done in 2
+   business days (excluding holidays in the USA and Germany). If you do not get
+   a reviewer within that time frame, you can ask for one by sending a mail to
+   [bazel-dev@googlegroups.com](mailto:bazel-dev@googlegroups.com). You can also
+   assign yourself a reviewer if you know who the reviewer should be (e.g.,
+   because they reviewed an earlier related change).
+7. Complete a code review. For each change, create a new commit and push it to
+   make changes to your pull request.
+8. A Bazel maintainer at Google applies the patch to our internal version
```
Comment (katre) [inline contributing.md]: Most of the rest of this is written in future tense, so change this to "will apply the patch".

File: contributing.md (hunk 2 above)
Comment (jin) [reply contributing.md]: added a step, ptal.

Comment (jin) [reply contributing.md]: combined the step with the previous one.

**Pull request merged**